File: hello-aws-lambda-nodejs12.x/tests/fnHelloTests.js
"use strict";

const app = require("../fnHello.js");
const chai = require("chai");
const expect = chai.expect;

describe("processRequest works when throwError is false", function () {
    it("testProcessRequestNoError", () => {
        const event = {
            "firstName": "Bogdan-Eugen",
            "lastName": "Mihai",
            "age": 34,
            "throwError": false
        }
        const response = app.exportedProcessRequest(event, "npm")

        expect(response).to.be.an("object");
        expect(response.greeting).to.be.equal("Hello Bogdan-Eugen Mihai, this is a Nodejs12.x AWS Lambda function developed using npm!");
        expect(response.result).to.be.equal("Your age in hexadecimal is 22");
    });
});

describe("processRequest throws error when throwError is true", function () {
    it("testProcessRequestThrowError", () => {
        const event = {
            "firstName": "Bogdan-Eugen",
            "lastName": "Mihai",
            "age": 34,
            "throwError": true
        }

        expect(function() {
            app.exportedProcessRequest(event, "npm")
        }).to.throw(Error).with.property("message", "Sorry, but the caller wants to me to throw an error");
    });
});

File: hello-aws-lambda-python3.8/tests/fn_hello_tests.py
import unittest

from app import fn_hello


class TestProcessRequest(unittest.TestCase):

    def test_process_request_no_error(self):

        event = {
            "first_name": "Bogdan-Eugen",
            "last_name": "Mihai",
            "age": 34,
            "throw_error": False
        }
        response = fn_hello.process_request(event, "pip")

        self.assertEqual("Hello Bogdan-Eugen Mihai, this is a Python3.8 AWS Lambda function developed using pip!",
                         response["greeting"])
        self.assertEqual("Your age in hexadecimal is 0x22", response["result"])

    def test_process_request_throw_error(self):

        event = {
            "first_name": "Bogdan-Eugen",
            "last_name": "Mihai",
            "age": 34,
            "throw_error": True
        }

        with self.assertRaises(Exception):
            fn_hello.process_request(event, "pip")


if __name__ == '__main__':
    unittest.main()
